### PR TITLE
HPCC-28010 Common up CMake handling for plugins

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: plugins/auditlib/lib_auditlib.ecllib [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export AuditLib := SERVICE
```
Comment (ghalliday) [inline plugins/auditlib/lib_auditlib.ecllib:1]: I think this needs a plugin attribute - so eclcc knows which .so to link in.

File: plugins/debugservices/lib_debugservices.ecllib [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export DebugServices := SERVICE
```
Comment (ghalliday) [inline plugins/debugservices/lib_debugservices.ecllib:1]: Also missing plugin attribute (and in other places below)

File: plugins/auditlib/CMakeLists.txt [hunk 3]
```diff
@@ -24,20 +24,32 @@
...
-
-HPCC_ADD_LIBRARY( auditlib SHARED ${SRCS} )
-install ( TARGETS auditlib DESTINATION plugins )
-target_link_libraries ( auditlib  
-         jlib 
-    )
-
+if(AUDITLIB)
+    ADD_PLUGIN(auditlib)
+    if(MAKE_AUDITLIB)
```
Comment (ghalliday) [inline plugins/auditlib/CMakeLists.txt:29]: question: What is the difference between AUDITLIB and MAKE_AUDITLIB?

**Review by ghalliday — COMMENTED** [review]
@GordonSmith a few initial comments

File: plugins/auditlib/lib_auditlib.ecllib (hunk 1 above)
Comment (GordonSmith) [reply plugins/auditlib/lib_auditlib.ecllib]: Resolved

File: plugins/debugservices/lib_debugservices.ecllib (hunk 2 above)
Comment (GordonSmith) [reply plugins/debugservices/lib_debugservices.ecllib]: Resolved

File: plugins/auditlib/CMakeLists.txt (hunk 3 above)
Comment (GordonSmith) [reply plugins/auditlib/CMakeLists.txt:29]: It seemed to be the common pattern, The MACROs allow you to exclude all except the one you want to include or include all except the ones you suppress.

**Review by ghalliday — APPROVED** [review]
I don't see any obvious problems.  @GordonSmith needs rebasing

**Pull request closed without merging**